apps/backend/src/libs/modules/controller/base-controller.module.ts

```diff
@@ -49,6 +53,24 @@ class BaseController implements Controller {
 		return await reply.status(status).send(payload);
 	}
 
+	private mapPreHandlers(
+		preHandlers: APIPreHandler[],
+	): ((
+		request: FastifyRequest,
+		reply: FastifyReply,
+		done: () => void,
+	) => void)[] {
+		return preHandlers.map((preHandler) => {
+			return (
+				{ body, params, query, user }: FastifyRequest,
+				_: FastifyReply,
+				done: () => void,
+			): void => {
+				preHandler({ body, params, query, user }, done);
+			};
```

Comment:
We already have a mapper for the request, let's use it since it's returning the same interface
```suggestion
			return (
				request: FastifyRequest,
				_: FastifyReply,
				done: () => void,
			): void => {
			    const preHandlerOptions = this.mapRequest(request);
			    
				preHandler(preHandlerOptions, done);
			};
```